ui/src/test/java/edu/wpi/grip/ui/codegeneration/PipelineGenerator.java

```diff
@@ -0,0 +1,45 @@
+package edu.wpi.grip.ui.codegeneration;
+
+import java.io.File;
+import java.nio.file.Paths;
+import javax.inject.Inject;
+import com.google.common.eventbus.EventBus;
+
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.events.ConnectionAddedEvent;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.Connection;
+import edu.wpi.grip.core.OperationMetaData;
+import edu.wpi.grip.core.util.ExceptionWitness;
+import edu.wpi.grip.core.util.MockExceptionWitness;
+public class PipelineGenerator {
+	@Inject
+	private Pipeline pipeline;
+	@Inject
+	private Exporter exporter;
+	@Inject
+	private EventBus eventBus;
+	@Inject
+	private Connection.Factory<Object> factory;
+	public PipelineGenerator(){
+		
+	}
+	final File codeDir = Paths.get("ui", "src","test","java","edu","wpi","grip","ui","codegeneration").toFile();
```

Comment:
`Paths.get` gets a path relative to the directory that the project is built from, even when running from a built `.jar`. So if you've cloned GRIP to `/Users/foo/Documents/GitHub/GRIP/`, it'll try to get the path `/Users/foo/Documents/GitHub/GRIP/ui/src/....`, which is very fragile. Since you're just trying to get a resource file that you know the name of, use `getClass().getResource()` to get the file